### PR TITLE
refactor: `partition_pdf()` pass `kwargs` through `fast` strategy pipeline

### PR DESCRIPTION
**Pull request by christinestraub**
This PR aims to pass `kwargs` through `fast` strategy pipeline, which was missing as part of the previous PR - https://github.com/Unstructured-IO/unstructured/pull/3030. 
I also did some code refactoring in this PR, so I recommend reviewing this PR commit by commit.

### Summary
- pass `kwargs` through `fast` strategy pipeline, which will allow users to specify additional params like `sort_mode`
- refactor: code reorganization
- cut a release for `0.14.0`
### Testing
CI should pass

### CODE REVIEW

File: unstructured/partition/pdf.py [hunk 1]
```diff
@@ -352,19 +351,133 @@ def extractable_elements(
...
-    filename: Optional[str] = "",
-    date_from_file_object: bool = False,
-) -> str | None:
-    last_modification_date = None
-    if not file and filename:
-        last_modification_date = get_last_modified_date(filename=filename)
-    elif not filename and file:
-        last_modification_date = (
-            get_last_modified_date_from_file(file) if date_from_file_object else None
+def _partition_pdf_with_pdfminer(
```
Comment (MthwRobinson) [inline unstructured/partition/pdf.py:354]: @christinestraub - Do you know why the diff for this file is so big?

Comment (christinestraub) [reply unstructured/partition/pdf.py:354]: It's because I did some code refactoring in this PR.

Comment (MthwRobinson) [reply unstructured/partition/pdf.py:354]: Sounds good! Will go through all the commits this morning.

**Review by MthwRobinson — APPROVED** [review]
Updates and refactor look good. Just one quick question, otherwise LGTM if tests are passing!

File: unstructured/partition/pdf_image/pdf_image_utils.py [hunk 2]
```diff
@@ -361,3 +368,52 @@ def annotate_layout_elements(
...
         else:
             raise FileNotFoundError(f'File "{filename}" not found!') from e
+
+
+def convert_pdf_to_images(
+    filename: str = "",
+    file: Optional[bytes | IO[bytes]] = None,
+    chunk_size: int = 10,
+) -> Iterator[Image.Image]:
+    # Convert a PDF in small chunks of pages at a time (e.g. 1-10, 11-20... and so on)
```
[the remark this answers is not in the document]
Comment (christinestraub) [reply unstructured/partition/pdf_image/pdf_image_utils.py:378]: Converting a PDF to images page by page can consume a significant amount of memory, especially if the PDF contains a high number of pages. By processing in chunks, we can manage memory usage more efficiently, reducing the risk of running out of memory.

**Pull request merged**